Update EIP-1: Expand External Link Policy (#5474)
ethereum/EIPs: EIPS/eip-1.md

## Linking to External Resources

> External resources are resources that are hosted outside of the EIPs repository. Links to external resources **MUST NOT** be included, except in `Living`, `Meta`, or `Informational` EIPs. External resources may disappear, move, or change unexpectedly. 

Reviewer: ```suggestion
Links to external resources **SHOULD NOT** be included without justification over and mitigation against the following caveats: 
1. External resources may disappear, move, or change unexpectedly. Therefore, EIP should choose permlink or archive link over transient links/
2. EIP is not meant for project-promotional purpose. Links to reference implementations or mentioning of projects outside of @ethereum org's repo MUST be justified based on merit by a consensus amongst Editor group .
```

> The following **MAY** be allowed at editor discretion:

Reviewer: ```suggestion
The following **MAYBE** included without explicit justification:
```
In addition to objection to change SHOULD NOT to MUST NOT
I also *object* to the term `at editor discretion` because it give editors too much power.

> - Links to `ethereum/go-ethereum` commits

Reviewer: ```suggestion
- Links to **ANY** `ethereum/<repo>` permlinks
```

Author: No.

- Links to `ethereum/execution-specs` commits
- Links to `ethereum/consensus-specs` commits

> - Resources with an appropriate backup in the assets folder

Reviewer: ```suggestion
- Links to any standard body existence over 10y, such as IETF, ISO, NIST, W3C,...
- Resources with an appropriate backup in the assets folder
```

Author: This is outside the scope of this PR.

## Linking to External Resources

External resources are resources that are hosted outside of the EIPs repository. External resources may disappear, move, or change unexpectedly. The following rules apply:

- Links in `Living`, `Meta`, or `Informational` EIPs are allowed
- Links to exact commits hosted on GitHub **MUST** be allowed if they are from one of the following repositories:
  - `ethereum/EIPs`
  - `ethereum/execution-specs`
  - `ethereum/consensus-specs`
- Links to Ethereum Magicians **MUST** be allowed in the `discussions-to` preamble
- Links with an appropriate backup in the assets folder **MAY** be allowed

> - Links to external resources **SHOULD NOT** be included if one of the above rules does not apply

Author: I can accept this.

> If you feel that a link you wish to include shouldn't violate the above rules, please submit a Pull Request to change them. Remember that the smaller and more specific the change, the more likely it will be accepted.

Author: > PRs are always welcome.

***People don't realize that.*** You wouldn't believe the number of comments like "why is such and such a rule a thing, I would like to change it to XYZ." People are afraid of touching things that they think are hallowed or something. Stating explicitly that it is not sacred aids in removing that barrier.

## Linking to External Resources

External resources are resources that are hosted outside of the EIPs repository. External resources may disappear, move, or change unexpectedly. The following rules apply in order.

If you feel that a link you wish to include shouldn't violate the below rules, please submit a Pull Request to change them. Remember that the smaller and more specific the change, the more likely it will be accepted.

### Access Rules

- All external resources **MUST** be free to both access and distribute.
- All external resources **MUST** use one of the following whitelisted URI schemes:
  - `https`
  - `ipfs`

### Reference Rules

> - All external resources **MUST** be non-normative.

Reviewer 2: -1 to this. One of the main reasons for addressing links in the first place would be to normatively reference other documents. Without this ability links still remain a blocker for 5593.

Author: Fair enough.
